### ISSUE TITLE
Fine-tune ConvNext

### ISSUE THREAD

**Issue by jedrzejwalega** [Appsilon/image_flow_cytometry_fine_tune#6]
Now that we've [reproduced](https://github.com/orgs/Appsilon/projects/154/views/1?pane=issue&itemId=87598804&issue=Appsilon%7Cimage_flow_cytometry_fine_tune%7C4) the authors' ResNet18 results (mostly to confirm the `lightning` and `neptune` framework works same way as their `skorch`) we want to try more recent models than ResNet.

The goal is to fine-tune ConvNext (pretrained can be taken from `timm`) on our [dataset](https://console.cloud.google.com/storage/browser/image-flow-cytometry-finetune/metadata?inv=1&invt=AbiAZQ&project=marekresearch&pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))).

We are benchmarking against the authors' val F1 = 0.91.

**Comment by jedrzejwalega** [comment]
First trial run for ConvNext [here](https://app.neptune.ai/o/appsilon/org/image-flow-cytometry-finetune/runs/details?viewId=standard-view&detailsTab=charts&shortId=IM-18&type=run).

Lessons learned:

- The `learning_rate` is way too big, around step = 15k we can see the moment ConvNext starts to harshly overfit in a giant drop of `val_accuracy`
- 500 `epochs` is way too much. To launch a lot of experiments I'll go with 5-10 `epochs` from now on.  

To find a more reasonable learning rate I'll launch a training with the model's recommended lr and implement `fastai` `lr_finder` for future usage, while the model trains.

![Image](https://github.com/user-attachments/assets/cc657064-ed8a-4b2d-a12a-080e48a24962)

**Comment by jedrzejwalega** [comment]
Learning rate finder implementation [here](https://github.com/Appsilon/image_flow_cytometry_fine_tune/blob/finetune/6%20-%20Appsilon/1%20-%20Machine%20learning/Fine-tuning%20ConvNext/Convnext_find_lr.py)

**Comment by jedrzejwalega** [comment]
Picked `learning_rate = 0.001` for a shorter training (actually went for more epochs than assumed above, that's due to the training launching before the decision was made). Results [here](https://app.neptune.ai/o/appsilon/org/image-flow-cytometry-finetune/runs/details?viewId=standard-view&detailsTab=charts&shortId=IM-19&type=run).

The `learning_rate = 0.0004` it started with seemed to work well for the model, so I've followed up with another run with this particular `learning_rate` [here](https://app.neptune.ai/o/appsilon/org/image-flow-cytometry-finetune/runs/details?viewId=standard-view&detailsTab=charts&shortId=IM-20&type=run).

Major observation: **scheduler seems to be broken**. It's supposed to be a OneCycleLR scheduler, yet it does not do full cycles. Need to fix before movign forward.

**Comment by jedrzejwalega** [comment]
Scheduler [fixed](https://github.com/orgs/Appsilon/projects/154/views/1?pane=issue&itemId=88015456&issue=Appsilon%7Cimage_flow_cytometry_fine_tune%7C5), can progress with the fine-tuning.

**Comment by jedrzejwalega** [comment]
[Here](https://github.com/orgs/Appsilon/projects/154/views/1?pane=issue&itemId=88016082&issue=Appsilon%7Cimage_flow_cytometry_fine_tune%7C8) we have found out that simply modifying authors' approach is probably not enough to beat their results.

We will design a completely new solution, starting from augmentations. The ConvNext idea does not go away, but will not be used with the authors' established approach as above.
